Core/GDCore/Project/ObjectsContainer.cpp

```diff
@@ -21,6 +22,22 @@ ObjectsContainer::ObjectsContainer() {
 
 ObjectsContainer::~ObjectsContainer() {}
 
+ObjectsContainer::ObjectsContainer(const ObjectsContainer& other) {
+  Init(other);
+}
+
+ObjectsContainer& ObjectsContainer::operator=(
+    const ObjectsContainer& other) {
+  if (this != &other) Init(other);
+
+  return *this;
+}
+
+void ObjectsContainer::Init(const gd::ObjectsContainer& other) {
+  initialObjects = gd::Clone(other.initialObjects);
+  objectGroups = other.objectGroups;
+}

```

Comment:
What about `rootFolder`?

Reply:
In the current release, the copy constructor of `Layout` don't copy objects at all. I guess, it works because the UI do a serialize + unserialize instead of a clone.

Here, it copies the objects but not the folders. It's a bit better.
The object tree sound hard to clone as it would need a map from the old objects to the new objects. I'm not sure it worth the effort.
Maybe I should just add a comment?

Reply:
You can add a comment but I also think that rootFolder should then be "reseted" because otherwise there is a risk it points to objects not owned by the container, leading to Very Bad Things.